### PR TITLE
WebGL / Implement the `line-metric` operator to apply styles/filter along a line

### PR DESCRIPTION
**Pull request by romprokop**
This Pull Request implements the `['line-metric']` operator, which returns the M value alongside a LineString. This can be used to filter parts of a line, change its color, etc. When no M component is available (e.g. the LineString has an XY or XYZ layout), the operator returns 0.

To implement this operator the following changes were made in the WebGL rendering pipeline:
* instead of storing LineString coordinates as [x, y] in the WebGL buffers, we store them as [x, y, m]
* the default LineString shaders were adapted to read the new M component, which is interpolated between vertices of a LineString
* because of the way the LineString buffers are built, the memory consumption for these geometries is increased by up to 20-25%

A new example as well as a rendering test were added, see:
https://deploy-preview-15961--ol-site.netlify.app/en/latest/examples/filter-webgl-line.html

Peek 31-07-2024 12-12

Todo:
- [x] fix rendering test (failing)
- [x] improve the example

Done in pair-programming with @jahow 
Follow-up of https://github.com/openlayers/openlayers/discussions/15790

### CODE REVIEW

**Comment by jahow** [comment]
This is now ready for review, please let us know if anything more is required to help with reviewing. Thanks a lot in advance

**Review by fredj — APPROVED** [review]

File: test/browser/spec/ol/webgl/styleparser.test.js [hunk 1]
```diff
@@ -325,6 +325,7 @@ describe('ol/webgl/styleparser', () => {
           };
           uid = computeHash(style['icon-src']);
           result = parseLiteralStyle(style);
+          await new Promise((resolve) => setTimeout(resolve, 500));
```
Comment (jahow) [inline test/browser/spec/ol/webgl/styleparser.test.js:328]: to remove

File: test/rendering/cases/webgl-line-metric/main.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  target: 'map',
+  view: new View({
+    center: [0, 0],
+    resolution: 1,
+    zoom: 1,
+  }),
+});
+
+render({
+  message: 'renders two lines side-by-side, one closed and one open',
```
Comment (jahow) [inline test/rendering/cases/webgl-line-metric/main.js:105]: to adjust

File: examples/data/geojson/switzerland-m-metric.geojson [hunk 3]
```diff
@@ -0,0 +1,47 @@
+{
+  "type": "FeatureCollection",
```
Comment (jahow) [inline examples/data/geojson/switzerland-m-metric.geojson:2]: to remove

File: src/ol/render/webgl/MixedGeometryBatch.js [hunk 4]
```diff
@@ -317,9 +321,10 @@ class MixedGeometryBatch {
    * @param {Feature|RenderFeature} feature Feature
    * @param {string} featureUid Feature uid
    * @param {number} stride Stride
+   * @param {string} [layout] Layout
```
Comment (jahow) [inline src/ol/render/webgl/MixedGeometryBatch.js:324]: improve typing

File: src/ol/geom/flat/transform.js [hunk 5]
```diff
@@ -9,6 +9,7 @@
  * @param {number} stride Stride.
  * @param {import("../../transform.js").Transform} transform Transform.
  * @param {Array<number>} [dest] Destination.
+ * @param {number} [destStride] DestStride.
```
Comment (jahow) [inline src/ol/geom/flat/transform.js:12]: to clarify

**Comment by jahow** [comment]
thanks for the review @fredj !!

**Pull request merged**